src/Map/index.tsx

```diff
@@ -81,6 +82,34 @@ const applyLayerDefaults = (props: Props) => {
   }
 }
 
+const updateMap = (sourceJson: GeoJSON, map: mapboxgl.Map) => {
+  if ((sourceJson as FeatureCollection).features.length > 0) {
+    let bounds = (sourceJson as FeatureCollection).features.reduce(function (
+      bounds: any,
+      feature: any
+    ) {
+      if (!Array.isArray(feature.geometry.coordinates[0])) {
+        return bounds.extend(feature.geometry.coordinates)
+      } else {
+        return feature.geometry.coordinates.reduce(function (
+          bounds: any,
+          coord: any
+        ) {
+          return bounds.extend(coord)
+        },
+        bounds)
+      }
+    },
+    new mapboxgl.LngLatBounds())
+    map.fitBounds(bounds, {
+      linear: false,
+      maxZoom: 12,
+      speed: 0.6,
+      padding: 100,
+    })
+  }
+}
```

Comment:
This looks like a potentially complicated calculation. Can I recommend calculating `bounds` in a `useMemo` and then using that in the effect to call `map.fitBounds`?